src/main/java/io/fabric8/jenkins/openshiftsync/JobProcessor.java

```diff
@@ -160,26 +186,32 @@ private void populateNamespaceFolder(Jenkins activeInstance, ItemGroup parent, S
 	}
 
 	private String populateBCProjectProperty(WorkflowJob job, String existingBuildRunPolicy,
-			BuildConfigProjectProperty buildConfigProjectProperty) throws IOException {
+			BuildConfigProjectProperty buildConfigProjectProperty) throws Exception {
 		if (buildConfigProjectProperty != null) {
+
 			existingBuildRunPolicy = buildConfigProjectProperty.getBuildRunPolicy();
+
 			long updatedBCResourceVersion = parseResourceVersion(buildConfig);
 			long oldBCResourceVersion = parseResourceVersion(buildConfigProjectProperty.getResourceVersion());
+
 			BuildConfigProjectProperty newProperty = new BuildConfigProjectProperty(buildConfig);
+
 			if (updatedBCResourceVersion <= oldBCResourceVersion
 					&& newProperty.getUid().equals(buildConfigProjectProperty.getUid())
 					&& newProperty.getNamespace().equals(buildConfigProjectProperty.getNamespace())
 					&& newProperty.getName().equals(buildConfigProjectProperty.getName())
 					&& newProperty.getBuildRunPolicy().equals(buildConfigProjectProperty.getBuildRunPolicy())) {
 				return null;
 			}
-			buildConfigProjectProperty.setUid(newProperty.getUid());
-			buildConfigProjectProperty.setNamespace(newProperty.getNamespace());
+
+      updateUID(buildConfigProjectProperty, newProperty);
+      buildConfigProjectProperty.setNamespace(newProperty.getNamespace());
```

Comment:
Maybe we need a test to surround this one. Also, because the namespace name is contained in the directory name inside of the Jenkins volume /var/lib/jenkins/<namespacename>-<buildconfigname>

Maybe, in some cases, we will need to rename the directory.